### PR TITLE
Do not throw deprecation warning if deprecated property is not presented in configuration

### LINKED ISSUE #5791: Having any (non-deprecated) `opening_brace` option in `.swiftlint.yml` triggers deprecation warning
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

If you include any `opening_brace` option in `.swiftlint.yml`, a warning is emitted right at the beginning of swiftlint's output:

```
% swiftlint
warning: Configuration option 'allow_multiline_func' in 'opening_brace' rule is deprecated. Use the option 'ignore_multiline_function_signatures' instead.
Linting Swift files in current working directory
Linting 'SwiftLintDemoApp.swift' (1/2)
Linting 'ContentView.swift' (2/2)
Done linting! Found 0 violations, 0 serious in 2 files.
```

### Environment

* SwiftLint version 0.57.0
* Installation method used: Homebrew
* Example configuration file:
```yml
opening_brace:
  ignore_multiline_statement_conditions: true
  ignore_multiline_function_signatures: true
```

### PR DESCRIPTION
**Pull request by chipp**
Fixes #5791

### CODE REVIEW

**Comment by chipp** [comment]
@SimplyDanny, could you check that and potentially release 0.57.1 hotfix? Thank you!

File: Tests/SwiftLintFrameworkTests/RuleConfigurationDescriptionTests.swift [hunk 1]
```diff
@@ -499,6 +499,15 @@ final class RuleConfigurationDescriptionTests: XCTestCase {
         )
     }
 
+    func testNoDeprecationWarningIfNoDeprecatedPropertySet() throws {
+        var configuration = TestConfiguration()
+
+        XCTAssertEqual(
+            try Issue.captureConsole { try configuration.apply(configuration: ["flag": false]) },
+            ""
+        )
```
Comment (SimplyDanny) [inline Tests/SwiftLintFrameworkTests/RuleConfigurationDescriptionTests.swift:508]:
```suggestion
        XCTAssert(try Issue.captureConsole { try configuration.apply(configuration: ["flag": false]) }.isEmpty)
```

File: Source/SwiftLintCoreMacros/RuleConfigurationMacros.swift [hunk 2]
```diff
@@ -74,7 +74,9 @@ enum AutoConfigParser: MemberMacro {
                 """
                 for option in nonInlinedOptions {
                     """
-                    try \(raw: option).apply(configuration[$\(raw: option).key], ruleID: Parent.identifier)
+                    if configuration.keys.contains($\(raw: option).key) {
+                        try \(raw: option).apply(configuration[$\(raw: option).key], ruleID: Parent.identifier)
+                    }
```
Comment (SimplyDanny) [inline Source/SwiftLintCoreMacros/RuleConfigurationMacros.swift:79]:
```suggestion
                    if let value = configuration[$\(raw: option).key] {
                        try \(raw: option).apply(value, ruleID: Parent.identifier)
                    }
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,10 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#5787](https://github.com/realm/SwiftLint/issues/5787)
 
+* Do not throw deprecation warning if deprecated property is not presented in configuration.
```
Comment (SimplyDanny) [inline CHANGELOG.md:22]:
```suggestion
* Do not throw deprecation warning if deprecated property is not
  presented in configuration.  
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,10 @@
   [SimplyDanny](https://github.com/SimplyDanny)
   [#5787](https://github.com/realm/SwiftLint/issues/5787)
 
+* Do not throw deprecation warning if deprecated property is not presented in configuration.  
+  [chipp](https://github.com/chipp)
+  [ #5791](https://github.com/realm/SwiftLint/issues/5791)
```
Comment (SimplyDanny) [inline CHANGELOG.md:24]:
```suggestion
  [#5791](https://github.com/realm/SwiftLint/issues/5791)
```

**Review by SimplyDanny — APPROVED** [review]
Apart from a few nits this looks good to me. Thanks!

File: Source/SwiftLintCoreMacros/RuleConfigurationMacros.swift (hunk 2 above)
Comment (chipp) [reply Source/SwiftLintCoreMacros/RuleConfigurationMacros.swift:79]: Shouldn't we support overwriting the default value with null here?

Comment (SimplyDanny) [reply Source/SwiftLintCoreMacros/RuleConfigurationMacros.swift:79]: Passing `nil` is a no-op right now and with the change this behavior will be retained.

**Pull request merged**